Classes/Utility/Parsers/DiffParser.swift

```diff
@@ -0,0 +1,223 @@
+//
+//  DiffParser.swift
+//  Freetime
+//
+//  Created by Weyert de Boer on 14/10/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+extension String {
+
+    func components(from matches: [NSTextCheckingResult]) -> [[String]] {
+        let line = self
+        return matches.map { result in
```

Comment:
I'm sorry I genuinely can't read the code in the function, Could we break it up a little to make a bit more sense please?

Reply:
Gets the strings from all the groups of a regular expression match :)